resource-managers/nomad/src/main/scala/org/apache/spark/scheduler/cluster/nomad/NomadJobManipulator.scala

```diff
@@ -58,6 +58,14 @@ private[spark] class NomadJobManipulator(val nomad: NomadScalaApi, private var j
     }
   }
 
+  def stopAlloc(allocId: String): Unit = {
+    nomad.allocations.signal(allocId, "SIGINT")
```

Comment:
we are signaling the allocation here rather than doing an explicit `nomad alloc stop` operation because Nomad does not honor the task group's restart policy when stopping an allocation.  We need the delay configured in the restart policy so we can properly wait for the alloc to stop and then lower the task count to avoid both 1) the removal of an allocation we don't want and 2) the immediate restart of an allocation we're trying to remove